src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -931,14 +932,15 @@ int doFinal(byte[] input, int inputOffset, int inputLen, byte[] output,
             // content of cipher has to be preserved in order for
             // users to retry the call with a larger buffer in the
             // case of ShortBufferException.
-            if (outputCapacity < estOutSize) {
+            if (output.length - outputOffset < estOutSize) {
                 cipher.save();
+                // create temporary output buffer if the estimated size is larger
+                // than the user-provided buffer.
+                internalOutput = new byte[estOutSize];
+                offset = 0;
             }
-            // create temporary output buffer so that only "real"
-            // data bytes are passed to user's output buffer.
-            outWithPadding = new byte[estOutSize];
         }
-        byte[] outBuffer = decrypting ? outWithPadding : output;
+        byte[] outBuffer = (internalOutput != null) ? internalOutput : output;
```

Comment:
With this line of change, the decrypted bytes are written into the output buffer including padding bytes. Is it not?

Reply:
If the output buffer provided is not big enough, it creates an internal one.  Two things happened here, one the variable was renamed because I felt "outWithPadding" was misleading.  The second was to only create this buffer when the length was smaller than the estimated length.  Before this change it would already create and use this buffer whether it was needed or not.

Reply:
I understand the renaming and etc. The difference here is that if output buffer is large enough, then the padding bytes are written into the output buffer even though the returned output length does not count them. Before this change, a separate output buffer is always used. Do you think it's ok to output padding bytes into user-supplied buffer? This is my main concern about this line.

Reply:
With me reverting padding ciphers to having a copy again, this should not be a concern.